query/task_manager.go

```diff
@@ -174,15 +173,14 @@ func (t *TaskManager) AttachQuery(q *influxql.Query, database string, interrupt
 	t.queries[qid] = query
 
 	go t.waitForQuery(qid, query.closing, interrupt, query.monitorCh)
-	if t.LogQueriesAfter != 0 {
+	if t.LogQueriesAfter != 0 && t.Diagnostic != nil {
```

Comment:
I know it's mostly obvious whats going on here, but it might be useful to note why we're checking if the diagnostic is nil here. I could see that as possibly being confusing if someone didn't have the context we do.